src/ProjectSystemTools/BuildLogging/Model/Backend/BackendBuildTableDataSource.cs

```diff
@@ -0,0 +1,127 @@
+{
+
+    [Export(typeof(IBackendBuildTableDataSource))]
+    internal sealed class BackendBuildTableDataSource : IBackendBuildTableDataSource // ITableEntriesSnapshotFactory, 
+    {
+        private const string BuildDataSourceDisplayName = "Build Data Source";
+        private const string BuildTableDataSourceIdentifier = nameof(BuildTableDataSourceIdentifier);
+        private const string BuildTableDataSourceSourceTypeIdentifier = nameof(BuildTableDataSourceSourceTypeIdentifier);
+
+        private readonly EvaluationLogger _evaluationLogger;
+        private readonly RoslynLogger _roslynLogger;
+
+        private ImmutableList<Build> _entries = ImmutableList<Build>.Empty;
+
+        public string SourceTypeIdentifier => BuildTableDataSourceSourceTypeIdentifier;
+
+        public string Identifier => BuildTableDataSourceIdentifier;
+
+        public string DisplayName => BuildDataSourceDisplayName;
+
+        public bool SupportRoslynLogging => _roslynLogger.Supported;
+
+        public bool IsLogging { get; private set; }
+
+        public bool SupportsRoslynLogging => _roslynLogger.Supported;
+
+        private NotifyCallback NotifyUI { get; set; }
+
+        public BackendBuildTableDataSource()
+        {
+            _evaluationLogger = new EvaluationLogger(this);
+            _roslynLogger = new RoslynLogger(this);
+        }
+
+        public void Start(NotifyCallback notifyCallback)
+        {
+            NotifyUI = notifyCallback;
+
+            IsLogging = true;
+            ProjectCollection.GlobalProjectCollection.RegisterLogger(_evaluationLogger);
+            _roslynLogger.Start();
+        }
+
+        public void Stop()
+        {
+            NotifyUI = null;
+
+            IsLogging = false;
+            ProjectCollection.GlobalProjectCollection.UnregisterAllLoggers();
+            _roslynLogger.Stop();
+        }
+
+        public void Clear()
+        {
+            foreach (var build in _entries)
+            {
+                build.Dispose();
+            }
+            _entries = ImmutableList<Build>.Empty;
+        }
+
+        public ILogger CreateLogger(bool isDesignTime) => new ProjectLogger(this, isDesignTime);
+
+        /// <summary>
+        /// (Temporary) return log path on server for a given build
+        /// If buildID cannot be found, will return null
+        /// </summary>
+        /// <param name="buildID">ID to return build for</param>
+        /// <returns>(Temporary) returns filepath to log path (on server)</returns>
+        public string RetrieveLogForBuild(int buildID)
+        {
+            Build match = findBuildByID(buildID);
+            return match.LogPath;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="buildID">ID to return build for</param>
+        /// <returns>Build matching the given ID, if no match, return null</returns>
+        private Build findBuildByID(int buildID)
```

Comment:
This method's name should start with a capital letter. Also, it's only used within the prior method so consider inlining it or making it a local function there.